test/Jamfile.v2

```diff
@@ -265,6 +265,7 @@ run allocator_destroy_test.cpp ;
 run allocator_construct_n_test.cpp ;
 run allocator_destroy_n_test.cpp ;
 run allocator_traits_test.cpp ;
+run allocator_pmr_test.cpp ;
```

Comment:
The test doesn't seem to execute any checks. Perhaps, it should be `compile` instead of `run`?

Reply:
Or maybe `link`.

Reply:
And while at it, if it doesn't need to execute any checks then it doesn't need to depend on `lightweight_test.hpp`.

Reply:
With `link` it looks like I still need a `main()` function, but I can leave it out with `compile`. I'll change it to `compile` to start